### PR TITLE
revise dataloder

### PR DESCRIPTION
**Pull request by FanGuozheng**
Revise the dataloader to make it more flexible.

### CODE REVIEW

File: tbmalt/io/dataset.py [hunk 1]
```diff
@@ -1,66 +1,109 @@
...
+                       targets: List[str],
+                       sizes: Union[List[int], int] = None,
+                       seed: int = 1):
+        """Load reference from h5py type data.
+
+        Arguments:
+            groups: The groups in h5py binary files. The input groups
+                can be a single group or a list of groups.
+            targets: Loading targets, such as dipole, charge, etc.
+            sizes: Loading size of each group.
```
Comment (mcsloy) [inline tbmalt/io/dataset.py:65]: Is `sizes` the size of each batch. If so we should probably call this `batch_sizes` just to make it clear what it is doing.

File: tbmalt/physics/dftb/__init__.py [hunk 2]
```diff
@@ -708,48 +710,45 @@ def reset(self):
...
+    basisb = Basis(geob.atomic_numbers, shell_dict={1: [0], 6: [0, 1], 8: [0, 1]})
+
+    mix_params = {'mix_param': 0.2, 'init_mix_param': 0.2, 'generations': 3, 'tolerance': 1e-10}
+    dftb2 = Dftb2(h_feed, s_feed, o_feed, u_feed, filling_temp=0.0036749324, mix_params=mix_params)
     dftb2(geos, basiss)
 
-    dftb2(geob, basisb)
-    assert torch.allclose(dftb2.q_final_atomic[1, :3], h2o)
-    assert torch.allclose(dftb2.q_final_atomic[0, :5], ch4)
-    dftb2(geob, basisb)
```
Comment (mcsloy) [inline tbmalt/physics/dftb/__init__.py:755]: We can probably remove the whole `if __name__ == '__main__':` section from this file

File: tbmalt/io/dataset.py [hunk 3]
```diff
@@ -1,66 +1,109 @@
...
+from h5py import Group
 import torch
 from torch.utils.data import DataLoader as Loader
 from torch import Tensor
 
-from tbmalt.common.batch import deflate, pack
+from tbmalt.common.batch import pack
 
 
 class Dataloader(Loader):
```
Comment (mcsloy) [inline tbmalt/io/dataset.py:13]: Might be best if the seed was an optional, rather than mandatory, argument.

Comment (mcsloy) [inline tbmalt/io/dataset.py:13]: We may also require some unit tests for the `Dataloader` class.

File: tbmalt/io/dataset.py [hunk 4]
```diff
@@ -1,66 +1,109 @@
...
-    """"""
+    """A class to read and load data.
 
-    def __init__(self, atomic_numbers: Tensor, positions: Tensor,
```
Comment (mcsloy) [inline tbmalt/io/dataset.py:16]: Some information about the structure of the arguments might be useful. I am unsure of what exactly is stored within `dataset`.

**Review by mcsloy — COMMENTED** [review]
Don't respond to these comments while you are on holiday.

**Comment by FanGuozheng** [comment]
@mcsloy Hi Adam, I have already merged your branch to this `io` branch, it looks good to me and it's ready to be merged to main branch :)
